src/js/components/Diagram/StyledDiagram.js

```diff
@@ -1,12 +1,82 @@
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 import { defaultProps } from '../../default-props';
+import { animationObjectStyle } from '../../utils/animations';
+
+const animationItemStyle = (animationType, theme) => {
+  if (typeof animationType === 'string') {
+    return animationObjectStyle({ type: animationType }, theme);
+  }
+  if (typeof animationType === 'object') {
+    return animationObjectStyle(animationType, theme);
```

Comment:
What happens if I call diagram with something like the following?
```
<Diagram animation={{ duration: 5000 }} />
```

With boolean, we default to 'draw'. Should we also default to 'draw' in this case? If so, should we account for this case in `animationStyle()`? Or is it on the caller to figure out?

Reply:
I'll test this out tonight as well, I actually didn't think of this! I think by default if not given a type in the object it should default to draw as it is implied that you want an animation by default.

Reply:
What are the findings of this thread?

Reply:
Both the connection level and component level animations will default to `draw` now if given an object but no specific `type` like the example Matt highlights in the thread.

Reply:
I think that if the type of `animation` is undefined it should default to how it is currently behaving.

Reply:
Just to clarify @ShimiSun, so should I keep it where it defaults to `draw` or if `animation` is undefined, nothing happens and the connection will instantly show up?

Reply:
As long as current users wouldn't have an added animation on the default behavior I think we should be good. Only if users specify `animation` prop it will default animation behavior to `draw`.